pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -313,6 +330,79 @@ func (n *networkManager) getGatewayDetails(svc *model.Service) []model.NetworkGa
 	return nil
 }
 
+func (n *networkManager) watchGatewayResources(c *Controller, stop <-chan struct{}) {
+	n.gatewayResourceClient = kclient.NewDelayedInformer(c.client, gvr.KubernetesGateway, kubetypes.StandardInformer, kubetypes.Filter{
+		LabelSelector: label.TopologyNetwork.Name,
+		FieldSelector: "spec.gatewayClassName=istio",
```

Comment:
I don't think we should or need to add this restriction. 

A user may use other Gateway classes - as long as they provide a TLS pass through, we should not 
restrict it to istio class.